libtiledbsoma/src/soma/soma_dataframe.cc

```diff
@@ -105,19 +105,37 @@ SOMADataFrame::SOMADataFrame(
         "auto",  // batch_size,
         result_order,
         timestamp);
-    array_->reset();
 }
 
 void SOMADataFrame::open(
     OpenMode mode, std::optional<std::pair<uint64_t, uint64_t>> timestamp) {
     array_->open(mode, timestamp);
-    array_->reset();
 }
 
 void SOMADataFrame::close() {
     array_->close();
 }
 
+bool SOMADataFrame::exists(std::string_view uri) {
```

Comment:
this seems like the kind of thing that could be pushed down more into C++ (not on this PR, which is big enough already) :)